File: src/main/java/io/pivotal/pivmart/products/Product.java
package io.pivotal.pivmart.products;

import lombok.Builder;
import lombok.Data;

import java.util.UUID;

@Data
@Builder
public
class Product {
    private UUID id;
    private UUID catalogId;
    private String name;
    private String price;
    private String description;
    private String imageSrc;
    private String imageAlt;
}

File: pivmart/frontend/src/domain/product/Product.ts
export interface Product {
  id: string
  name: string,
  price: string,
  description: string,
  imageSrc: string,
  imageAlt: string
}

export class ProductModel implements Product {
  public id: string = '';

  constructor(public name: string,
              public price: string,
              public description: string,
              public imageSrc: string,
              public imageAlt: string) {
  }
}
